### PR TITLE
[Merged by Bors] - Enable logging

### PR DESCRIPTION
**Pull request by maltesander**
# Description

fixes https://github.com/stackabletech/trino-operator/issues/376

## TODO

- [x] Release new operator-rs
- [x] Adapt cargo.toml to new operator-rs release

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/52/

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -5,8 +5,12 @@ use crate::{
     config::password_authenticator_properties,
 };
 
+use crate::product_logging::{
+    get_log_properties, get_vector_toml, resolve_vector_aggregator_address,
+};
 use indoc::formatdoc;
 use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::product_logging::spec::{ConfigMapLogConfig, CustomContainerLogConfig};
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: https://github.com/stackabletech/trino-operator/pull/380/commits/6557e0165c65502e7796c13bc5ed2d62ae169e74

File: rust/operator-binary/src/product_logging.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    Error,
+}
+
+impl From<LogLevel> for TrinoLogLevel {
+    fn from(level: LogLevel) -> Self {
+        match level {
+            LogLevel::TRACE | LogLevel::DEBUG => Self::Debug,
+            LogLevel::INFO => Self::Info,
+            LogLevel::WARN => Self::Warn,
+            LogLevel::ERROR | LogLevel::FATAL | LogLevel::NONE => Self::Error,
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/product_logging.rs]: Good point: https://github.com/stackabletech/trino-operator/pull/380/commits/361534f7f01cc16d2d64f24f8573c2b0b228cdf3

File: tests/templates/kuttl/logging/01-install-trino-vector-aggregator.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      helm install trino-vector-aggregator vector
+      --namespace $NAMESPACE
+      --version 0.19.0
+      --repo https://helm.vector.dev
+      --values trino-vector-aggregator-values.yaml
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vector-aggregator-discovery
```
[the remark this answers is not in the document]
Comment (maltesander) [reply tests/templates/kuttl/logging/01-install-trino-vector-aggregator.yaml]: done https://github.com/stackabletech/trino-operator/pull/380/commits/d52d22a40d94fc77492e2449fece5a27028f67e9

File: rust/operator-binary/src/product_logging.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        source: stackable_operator::error::Error,
+        cm_name: String,
+    },
+    #[snafu(display("failed to retrieve the entry {entry} for ConfigMap {cm_name}"))]
+    MissingConfigMapEntry {
+        entry: &'static str,
+        cm_name: String,
+    },
+    #[snafu(display("crd validation failure"))]
+    CrdValidationFailure { source: stackable_trino_crd::Error },
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/product_logging.rs]: removed

File: rust/operator-binary/src/controller.rs [hunk 5]
```diff
@@ -678,12 +765,28 @@ fn build_rolegroup_statefulset(
         catalogs,
     )?;
 
+    // TODO: prepare default missing!!!
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: leftover, removed

File: docs/modules/usage_guide/nav.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
+* xref:index.adoc[]
```
[the remark this answers is not in the document]
Comment (maltesander) [reply docs/modules/usage_guide/nav.adoc]: https://github.com/stackabletech/trino-operator/pull/380/commits/b5d733a529dc07590f730b3fc5b6277b0f44bb5f

[the remark this answers is not in the document]
Comment (siegfriedweber) [reply docs/modules/usage_guide/nav.adoc]: `docs/modules/trino/partials/nav.adoc` must contain the the navigation entries:

```adoc
* xref:trino:getting_started/index.adoc[]
** xref:trino:getting_started/installation.adoc[]
** xref:trino:getting_started/first_steps.adoc[]
* xref:trino:configuration.adoc[]
* xref:trino:concepts.adoc[]
* xref:trino:usage_guide/index.adoc[]
** xref:trino:usage_guide/cluster.adoc[]
** xref:trino:usage_guide/configuration.adoc[]
** xref:trino:usage_guide/security.adoc[]
** xref:trino:usage_guide/monitoring.adoc[]
** xref:trino:usage_guide/log-aggregation.adoc[]
** xref:trino:usage_guide/query.adoc[]
** xref:trino:usage_guide/catalogs/index.adoc[]
*** xref:trino:usage_guide/catalogs/black-hole.adoc[]
*** xref:trino:usage_guide/catalogs/google-sheets.adoc[]
*** xref:trino:usage_guide/catalogs/hive.adoc[]
*** xref:trino:usage_guide/catalogs/iceberg.adoc[]
*** xref:trino:usage_guide/catalogs/tpcds.adoc[]
*** xref:trino:usage_guide/catalogs/tpch.adoc[]
```

`docs/modules/usage_guide/nav.adoc` must be deleted.

**Review by siegfriedweber — CHANGES_REQUESTED** [review]
The navigation of the documentation must be fixed. Everything else looks good to me.

**Comment by maltesander** [comment]
Last test hopefully: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/54/

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**